src/main/scala/org/apache/spark/sql/pulsar/PulsarSource.scala

```diff
@@ -63,10 +79,79 @@ private[pulsar] class PulsarSource(
     initialTopicOffsets
```

Comment:
We need to throw an exception from getOffset to make sure it is not getting called.

Reply:
Done